Framework/DataHandling/src/LoadFITS.cpp

```diff
@@ -150,170 +170,175 @@ void LoadFITS::exec() {
  * Load header(s) from FITS file(s) into FITSInfo header
  * struct(s). This is usually the first step when loading FITS files
  * into workspaces or anything else. In the simplest case, paths has
- * only one string and only one header struct is added in headers.
+ * only one string and only one header struct is added in
+ * headers. When loading multiple files, the parameters firstIndex and
+ * lastIndex make it possible to load the headers individually or by
+ * groups.
  *
  * @param paths File name(s)
  * @param headers Vector where to store the header struct(s)
  *
+ * @param firstIndex index of the first header to read, set to 0 to
+ * start reading headers from the first file.
+ *
+ * @param lastIndex index of the last header to read
+ *
  * @throws std::runtime_error if issues are found in the headers
  */
 void LoadFITS::doLoadHeaders(const std::vector<std::string> &paths,
-                             std::vector<FITSInfo> &headers) {
-  headers.resize(paths.size());
+                             std::vector<FITSInfo> &headers, size_t firstIndex,
+                             size_t lastIndex) {
+  for (size_t i = firstIndex; i <= lastIndex && i <= headers.size(); ++i) {
+    loadHeader(paths[i], headers[i]);
+  }
+}
 
-  for (size_t i = 0; i < paths.size(); ++i) {
-    headers[i].extension = "";
-    headers[i].filePath = paths[i];
+void LoadFITS::loadHeader(const std::string &filePath, FITSInfo &header) {
+  header.extension = "";
+  header.filePath = filePath;
 
-    // Get various pieces of information from the file header which are used
-    // to
-    // create the workspace
-    try {
-      parseHeader(headers[i]);
-    } catch (std::exception &e) {
-      // Unable to parse the header, throw.
-      throw std::runtime_error(
-          "Severe problem found while parsing the header of "
-          "this FITS file (" +
-          paths[i] +
-          "). This file may not be standard FITS. Error description: " +
-          e.what());
+  // Get various pieces of information from the file header which are used
+  // to
+  // create the workspace
+  try {
+    parseHeader(header);
+  } catch (std::exception &e) {
+    // Unable to parse the header, throw.
+    throw std::runtime_error(
+        "Severe problem found while parsing the header of "
+        "this FITS file (" +
+        filePath +
+        "). This file may not be standard FITS. Error description: " +
+        e.what());
+  }
+
+  // Get and convert specific MANDATORY standard header values which
+  // will are needed to know how to load the data: BITPIX, NAXIS,
+  // NAXISi (where i = 1..NAXIS, e.g. NAXIS2 for two axis).
+  try {
+    std::string tmpBitPix = header.headerKeys[m_headerBitDepthKey];
+    if (boost::contains(tmpBitPix, "-")) {
+      boost::erase_all(tmpBitPix, "-");
```

Comment:
could use `std::find` and `std::replace` but this is actually easier to read :+1: